docker/gaffer-kerberos/gaffer-integration-tests-krb/krb-entrypoint.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# Copyright 2020-2023 Crown Copyright
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+# Show verbose Hadoop Kerberos auth information
+if [ "$DEBUG" -eq 1 ]; then
+  export HADOOP_JAAS_DEBUG=true
+  export HADOOP_OPTS="-Djava.net.preferIPv4Stack=true -Dsun.security.krb5.debug=true -Dsun.security.spnego.debug"
+  echo "Debugging flag enabled (DEBUG=$DEBUG), additional Kerberos details will be printed"
+fi
+
+KEYTAB_PATH=/tmp/gaffer.keytab
+PRINCIPAL=gaffer/$(hostname)
+FULL_PRINCIPAL=gaffer/$(hostname)@GAFFER.DOCKER
+
+{
+echo "add_entry -password -p $PRINCIPAL -k 1 -e aes256-cts"; sleep 0.2
+echo $GAFFER_KRB_PASSWORD; sleep 0.2
+echo list; sleep 0.2
+echo "write_kt $KEYTAB_PATH"; sleep 0.2
+echo exit
+} | ktutil
+
+# Update store properties files to point to the location of the Accumulo store to test against:
+accumulo_instance=$(cat conf/store.properties | grep accumulo.instance | sed -e 's/.*=\(.*\)/\1/')
+accumulo_zookeepers=$(cat conf/store.properties | grep accumulo.zookeepers | sed -e 's/.*=\(.*\)/\1/')
+store_properties=$(find /tmp/gaffer/store-implementation/accumulo-store/src/test/resources -name *.properties | grep -v cache)
+
+for store in $store_properties; do
+sed -i'' -e "s/gaffer.store.class=\(.*\)Mini\(.*\)/gaffer.store.class=\1\2/" $store
+sed -i'' -e "s/accumulo.instance=.*/accumulo.instance=$accumulo_instance/" $store
+sed -i'' -e "s/accumulo.zookeepers=.*/accumulo.zookeepers=$accumulo_zookeepers/" $store
+echo accumulo.kerberos.principal=$FULL_PRINCIPAL >> $store
+echo accumulo.kerberos.keytab=$KEYTAB_PATH >> $store
+echo accumulo.kerberos.enable=true >> $store
+done
+
+# Needed for AddElementsFromHdfs tests
+cp /opt/hadoop/conf/core-site.xml /tmp/gaffer/store-implementation/accumulo-store/src/test/resources
+cp /opt/hadoop/conf/hdfs-site.xml /tmp/gaffer/store-implementation/accumulo-store/src/test/resources
+
+# Required for Hadoop to find its Native Libraries which Kerberos auth cannot work without
+cp /tmp/hadoop/native/lib* /usr/lib
```

Comment:
This should be:
```bash
cp -r /tmp/hadoop/native/ib* /usr/lib/
```

Reply:
As with the other comment, it will work the same unless `lib` is missing.

Reply:
That's fair you might still need the `-r` option tho

Reply:
The command copies all the files from `/tmp/hadoop/native` which start with "lib". We would want `-r` if "lib" was a folder. The Hadoop lib folder has a subdirectory with example code (doesn't start with "lib") which we don't need to copy which is why the whole directory isn't copied.

When running the container the command is expanded to this: `cp /tmp/hadoop/native/libhadoop.a /tmp/hadoop/native/libhadoop.so /tmp/hadoop/native/libhadoop.so.1.0.0 /tmp/hadoop/native/libhadooppipes.a /tmp/hadoop/native/libhadooputils.a /tmp/hadoop/native/libhdfs.a /tmp/hadoop/native/libhdfs.so /tmp/hadoop/native/libhdfs.so.0.0.0 /tmp/hadoop/native/libnativetask.a /tmp/hadoop/native/libnativetask.so /tmp/hadoop/native/libnativetask.so.1.0.0 /usr/lib/`